### ISSUE TITLE
The order of Host header when making a request

### ISSUE THREAD

**Issue by ekim7809** [http-kit/http-kit#521]
The IETF doc says the the order of headers in a request does not matter but I was lucky enough to experience a problem with this recently. I set user-agent string and a couple of other headers before making a get request and the server, which is not owned by me, responds with a 503 error because the "Host" header is not in the first line. My request goes Accept:, User-Agent: and then the Host:. This site had no problem with http-kit but it keeps responding with 503 after some update on its server.

Would there be any workaround to make the Host header go in the first line? Although this would be against the standard, I found some other Web sites have implemented this scheme to prevent Web scrapers.

Answers and suggestions would be highly appreciated.

Thank you.

**Comment by ekim7809** [comment]
Problem solved.

The site had nginx proxy environment setup for load balancing and my http-kit client (v2.6.0) was getting 503 error because it wasn't carrying SNI. I tried to use sni-client with the same version but somehow I failed. Then I switched to v2.7.0-beta3 and now it's working fine.

I'm sure it's a good move to make SNI capability as default as more sites require it now.

Many thanks to the contributors!
